packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -197,12 +264,49 @@ const BoldChunk = (chunks) => <Typography fontWeight="bold">{chunks}</Typography
 const SelectedEntriesModalContent = ({ onToggle, onConfirm, onRefresh }) => {
   const { formatMessage } = useIntl();
   const { selectedEntries, rows, isLoading, isFetching } = useTableContext();
+  const [isDialogOpen, setIsDialogOpen] = React.useState(false);
+  const [isConfirmButtonLoading, setIsConfirmButtonLoading] = React.useState(false);
+  const [rowsToDisplay, setRowsToDisplay] = React.useState([]);
+  const [entriesToPublish, setEntriesToPublish] = React.useState([]);
 
-  const selectedEntriesWithErrorsCount = rows.filter(
+  const selectedEntriesWithErrorsCount = rowsToDisplay.filter(
     ({ entity, errors }) => selectedEntries.includes(entity.id) && errors
   ).length;
   const selectedEntriesWithNoErrorsCount = selectedEntries.length - selectedEntriesWithErrorsCount;
 
+  const toggleDialog = () => setIsDialogOpen((prev) => !prev);
+
+  const handleConfirmBulkPublish = async () => {
+    try {
+      toggleDialog();
+      setIsConfirmButtonLoading(true);
+      // Should we move the bulk publish mutation to this component?
```

Comment:
Should be removed.

But I am wondering if we should just move the mutation here. Now that we have the onRefresh we should be able to handle the error and success states here and it would also give us the isLoading from the mutation.